datadog_checks_base/datadog_checks/checks/prometheus/base_check.py

```diff
@@ -159,12 +159,14 @@ def get_scraper(self, instance):
         scraper.extra_headers.update(instance.get("extra_headers", {}))
         # For simple values instance settings overrides optional defaults
         scraper.prometheus_metrics_prefix = instance.get("prometheus_metrics_prefix", default_instance.get("prometheus_metrics_prefix", ''))
-        scraper.label_to_hostname = instance.get("label_to_hostname", default_instance.get("prometheus_url", ""))
+        scraper.label_to_hostname = instance.get("label_to_hostname", default_instance.get("label_to_hostname", ""))
```

Comment:
We should fix the tests so we actually catch these issues.

Reply:
Good practice indeed! I added it, but leaving the other config params untested. Let me know if you think I should add them in the same act :-)